libs/vm-modules/tests/unit/math/tensor_tests.cpp

```diff
@@ -355,6 +356,34 @@ TEST_F(MathTensorTests, tensor_set_from_string)
   EXPECT_TRUE(gt.AllClose(tensor->GetTensor()));
 }
 
+TEST_F(MathTensorTests, tensor_copy_from_tensor)
+{
+  static char const *tensor_from_string_src = R"(
+    function main() : Tensor
+      var tensor_shape = Array<UInt64>(1);
+      tensor_shape[0] = 4u64;
+
+      var x = Tensor(tensor_shape);
+      var y = Tensor(tensor_shape);
+      x.fill(2.0fp64);
+      y = x.copy();
+
+      return y;
+
+    endfunction
+  )";
+
+  ASSERT_TRUE(toolkit.Compile(tensor_from_string_src));
+  Variant res;
+  ASSERT_TRUE(toolkit.Run(&res));
+
+  auto const                    tensor = res.Get<Ptr<fetch::vm_modules::math::VMTensor>>();
+  fetch::math::Tensor<DataType> gt({4});
+  gt.Fill(static_cast<DataType>(2.0));
+
+  EXPECT_TRUE(gt.AllClose(tensor->GetTensor()));
+}
```

Comment:
I assume that x and y are deep copies and not references to the same object - it would be good to have a test for this.